### PR TITLE
chore!: move to a mono repo structure and update packages

### PR DESCRIPTION
**Pull request by westeezy**
# v8

We have noticed many of our packages only use webpack, babel, and eslint from `grumbler-scripts`. This means they are pulling all the dependencies of karma/eslint/etc when they arent required. We also have often mentioned we wanted to potentially move to a different linting style after we move to TypeScript. This PR sets us up so `grumbler-scripts` becomes a collection of config packages but still exposes a convenient wrapper `@krakenjs/grumbler-scripts`. The `@krakenjs/grumbler-scripts` is a drop in replacement for v7 so should maintain backwards compatibility.

# Changes Made:

## Breaking Changes:

- Minimum Node version is now v16 officially
- Remove `jest`, `babel-jest`, and `cross-env` from dependencies.
- Allow warning comments now such as `// TODO:`
- Increase `max-lines` limit to 1000
- Rewrite TypeScript Eslint rules to match [xo](https://github.com/xojs/xo)
- Change jsx settings in `tsconfig.json` from `preserve` to `react`
- Package major upgrades listed in #Package Upgrades below

## Non Breaking Changes:

- Move from semantic-release to changesets
- Split out configs into their own packages and utilize npm workspaces
- Add `package-lock.json` for deterministic installs

# Package Upgrades:

## Webpack:
All packages upgraded to latest version that supports webpack@4. Some
have more updates but dropped webpack@4 support in those updates.
 circular-dependency-plugin   ^5.0.2  →   ^5.2.2
 css-loader                   ^3.4.2  →   ^5.2.7
 isomorphic-style-loader      ^5.1.0  →   ^5.3.2
 raw-loader                   ^4.0.0  →   ^4.0.2
 sass                        ^1.51.0  →  ^1.55.0
 semver                       ^7.1.1  →   ^7.3.8
 terser-webpack-plugin        ^2.3.2  →   ^4.2.3
 text-loader                  [removed]
 webpack-cli                  ^3.3.12 →   ^4.10.0
 webpack-dev-server           ^3.1.14 →   ^4.11.1

 ## Babel:
 @babel/eslint-parser                     ^7.18.2  →  ^7.19.1
 @babel/node                               ^7.0.0  →  ^7.20.0
 @babel/plugin-proposal-class-properties  ^7.10.4  →  ^7.18.6
 @babel/plugin-proposal-decorators        ^7.10.5  →  ^7.20.0
 @babel/plugin-transform-runtime          ^7.11.0  →  ^7.19.6
 @babel/preset-flow                       ^7.10.4  →  ^7.18.6
 @babel/preset-typescript                  7.16.7  →   7.18.6
 @babel/register                          ^7.10.5  →  ^7.18.9
 @babel/runtime                           7.12.13  →   7.20.0
 babel-plugin-add-module-exports           ^1.0.0  →   ^1.0.4
 babel-plugin-istanbul                     ^6.0.0  →   ^6.1.1
 babel-plugin-react-scoped-css             ^1.0.0  →   ^1.1.1

## ESLint:
 eslint                              ^8.1.0  →  ^8.26.0
 eslint-import-resolver-typescript    2.7.1  →    3.5.2
 eslint-plugin-compat                ^4.0.0  →   ^4.0.2
 eslint-plugin-import               ~2.25.2  →  ~2.26.0
 eslint-plugin-promise               ^6.0.0  →   ^6.1.1
 eslint-plugin-react                 7.27.0  →  7.31.10
 eslint-plugin-security              ^1.4.0  →   ^1.5.0

# Changes Planned in the future:
- Webpack v5 or potentially investigating esbuild
- Revisiting of the eslint rules especially as they pertain to TypeScript
- Removing the build step for esm to cjs in webpack and karma config packages
- Revisiting the idea of consumer packages installing dependencies vs including them in these packages

### CODE REVIEW

**Review by wsbrunson — APPROVED** [review]

**Comment by wsbrunson** [comment]
We should probably update the README

**Review by wsbrunson — APPROVED** [review]

**Review by wsbrunson — APPROVED** [review]

**Comment by gregjopa** [comment]
This is awesome @westeezy!! Happy to approve after the conflicts are resolved.

File: README.md [hunk 1]
```diff
@@ -1,116 +1,57 @@
...
-```
-npm install @krakenjs/grumbler-scripts
-```
+This is a collection of configuration scripts one can use for their projects.
 
-Shared scripts for grumbler based modules.
+### Packages Available
 
-## Package.json scripts
+Below is a list of available packages to install. To provide a one stop shop we will continue to support `@krakenjs/grumbler-scripts` which is a combination of all the packages. Feel free to use that parent package or install configs as you need.
```
Comment (gregjopa) [inline README.md:7]: Great idea with still providing a single global package for ease of use while still allowing folks to pick and choose the features they want ❤️

**Review by gregjopa — APPROVED** [review]
💯

**Pull request merged**